### PR TITLE
fix(react): Small adjustments to some components.

### PR DESCRIPTION
**Pull request by ditman**
Adds a few classnames to some of the basic catalog components so they can be located in tests in a similar way to those on Lit and Angular: `Card`, `ChoicePicker`, `Modal` and `Tabs`.

Adds handling for the DateTimeInput value so it conforms with HTML5 values; similar to what was needed in the Lit package.

* Continues: #1607 
* Tests: #1625

## Pre-launch Checklist

- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I read the [Style Guide].
- [x] I have added updates to the [CHANGELOG].
- [ ] I updated/added relevant documentation.
- [x] My code changes (if any) have tests.
- [ ] If my branch is on fork, I have verified that [scripts/e2e_test.sh](../scripts/e2e_test.sh) passes.

If you need help, consider asking for advice on the [discussion board].

[CHANGELOG]: ../CHANGELOG.md
[CLA]: https://cla.developers.google.com/
[Contributors Guide]: ../CONTRIBUTING.md
[discussion board]: https://github.com/a2ui-project/a2ui/discussions
[Style Guide]: ../CONTRIBUTING.md#coding-style

### CODE REVIEW

File: renderers/react/src/v0_9/catalog/basic/components/DateTimeInput.tsx [hunk 1]
```diff
@@ -19,6 +19,26 @@ import {createComponentImplementation} from '../../../adapter';
...
+  switch (type) {
+    case 'date':
+      return datePart;
+    case 'time':
+      return timePart;
+    case 'datetime-local':
+      return `${datePart}T${timePart}`;
+  }
+  return '';
+}
```
[the remark this answers is not in the document]
Comment (josemontespg) [reply renderers/react/src/v0_9/catalog/basic/components/DateTimeInput.tsx:40]: Looks like it found an edge case, consider adding a test to cover this.

**Review by josemontespg — APPROVED** [review]
Makes sense, thank you for the improvement!

File: renderers/react/src/v0_9/catalog/basic/components/DateTimeInput.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ditman) [reply renderers/react/src/v0_9/catalog/basic/components/DateTimeInput.tsx:40]: ~Ah yes, I think I remember a similar comment on the Lit renderer, I'll try to apply whatever fix I did on Lit here so they (mis)behave the same :P~

[the remark this answers is not in the document]
Comment (ditman) [reply renderers/react/src/v0_9/catalog/basic/components/DateTimeInput.tsx:40]: OK, currently this is the implementation we have for Lit, and Angular has the same problem; so **for now** I'm not getting into the business of robustly attempting to parse date/time strings, which is a world of pain.

**Comment by ditman** [comment]
I ended up not getting in the business of parsing date/time strings. There's some libraries that would help us do this (Intl, any-date-parser) but for now, this is better than the implementation that is currently available (and is functionally equivalent to Lit and Angular)
